mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -258,6 +290,27 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  /** @group setParam */
+  @Since("2.0.0")
+  def setInitialModel(value: KMeansModel): this.type = set(initialModel, value)
```

Comment:
I think that there are certain params that have to be the same (k for
example) where overriding doesn't even make sense. So we can either print a
warning and continue with initial model params, or throw an error.

However overriding convergence tolerance or iterations, say, should be
possible and is certainly a feasible and valid user situation. In this case
we should support it.

We can document the behaviour accordingly and it will be a bit different
for each model...
On Tue, 15 Mar 2016 at 07:06, jkbradley <notifications@github.com> wrote:

> In mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala
> <https://github.com/apache/spark/pull/11119#discussion_r56115607>:
>
> > @@ -258,6 +290,27 @@ class KMeans @Since("1.5.0") (
> >    @Since("1.5.0")
> >    def setSeed(value: Long): this.type = set(seed, value)
> >
> > +  /** @group setParam */
> > +  @Since("2.0.0")
> > +  def setInitialModel(value: KMeansModel): this.type = set(initialModel, value)
>
> I think that's the user's problem if they overwrite k or other parameters
> after setting the initial model. However, I'm Ok with addressing this in
> the follow-up PR for overriding the Params when initialModel is set.
>
> —
> You are receiving this because you were mentioned. Reply to this email
> directly or view it on GitHub: <a href="
> https://github.com/apache/spark/pull/11119/files#r56115607">
> https://github.com/apache/spark/pull/11119/files#r56115607</a>
>